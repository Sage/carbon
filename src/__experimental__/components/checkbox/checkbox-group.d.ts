import * as React from 'react';
import { WithValidationProps } from '../../../components/validations/with-validation';

interface CheckboxGroupProps {
  children: React.FunctionComponent | React.ComponentClass;
  groupName: string;
  hasError: boolean;
  hasWarning: boolean;
  hasInfo: boolean;
}

declare const CheckboxGroup: React.ComponentClass<WithValidationProps & CheckboxGroupProps>;

export { CheckboxGroup };
