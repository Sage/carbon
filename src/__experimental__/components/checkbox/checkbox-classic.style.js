import { css } from 'styled-components';
import { StyledCheckableInput } from '../checkable-input/checkable-input.style';
import FieldHelpStyle from '../field-help/field-help.style';
import HiddenCheckableInputStyle from '../checkable-input/hidden-checkable-input.style';
import StyledCheckableInputSvgWrapper from '../checkable-input/checkable-input-svg-wrapper.style';
import LabelStyle from '../label/label.style';
import StyledHelp from '../../../components/help/help.style';
import { isClassic } from '../../../utils/helpers/style-helper';
import StyledIcon from '../../../components/icon/icon.style';

export default ({
  checked, disabled, fieldHelpInline, reverse, theme
}) => isClassic(theme) && css`
  ${StyledCheckableInput} {
    padding: 1px 0 0 0;
  }

  ${HiddenCheckableInputStyle},
  svg {
    padding: 1px;
  }

  ${StyledCheckableInput},
  ${HiddenCheckableInputStyle},
  ${StyledCheckableInputSvgWrapper},
  svg {
    height: 15px;
    width: 15px;
  }

  ${FieldHelpStyle} {
    margin-left: 15px;
  }

  ${LabelStyle} ${StyledHelp} {
    &, &:hover, &:focus {
      color: #8099a4;
    }
  }

  ${LabelStyle} ${StyledIcon}::before {
    font-size: 16px;
  }

  ${fieldHelpInline && `${FieldHelpStyle},`}
  ${LabelStyle} {
    padding-top: 0;
    padding-bottom: 0;
  }

  ${HiddenCheckableInputStyle}:not([disabled]) {
    &:focus + ${StyledCheckableInputSvgWrapper},
    &:hover + ${StyledCheckableInputSvgWrapper} {
      box-shadow: none;
    }
  }

  ${HiddenCheckableInputStyle}:not([disabled]) {
    &:focus + ${StyledCheckableInputSvgWrapper} > svg,
    &:hover + ${StyledCheckableInputSvgWrapper} > svg {
      border: 1px solid #1963f6;
      outline: none;
    }
  }

  ${checked && `
    svg path { fill: rgba(0, 0, 0, 0.85); }
  `}

  ${disabled && css`
    ${LabelStyle} {
      color: ${theme.disabled.text};
    }

    svg {
      background-color: #e6ebed;
      border-color: #bfccd2;
    }

    svg path { fill: #${checked ? '8099a4' : 'e6ebed'}; }
  `}

  ${(fieldHelpInline || reverse) && `
    ${FieldHelpStyle} {
      margin-left: 0;
    }
  `}
`;
