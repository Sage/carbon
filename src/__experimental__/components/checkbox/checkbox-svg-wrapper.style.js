import styled from 'styled-components';

const StyledCheckboxSvgWrapper = styled.div``;

export default StyledCheckboxSvgWrapper;
