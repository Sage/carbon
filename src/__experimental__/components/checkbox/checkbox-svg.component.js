import React from 'react';
import StyledCheckboxSvgWrapper from './checkbox-svg-wrapper.style';

const CheckboxSvg = () => {
  return (
    <StyledCheckboxSvgWrapper>
      <svg
        width='12'
        height='10'
        viewBox='0 0 12 10'
      >
        <path
          d={ 'M.237 6.477A.752.752 0 0 1 .155 5.47l.851-1.092a.63.63 0 0 1 .934-.088l2.697 1.964, '
            + '4.674-6a.63.63 0 0 1 .933-.088l1.015.917c.28.254.317.703.081 1.005L6.353 8.492a.725.725, '
            + '0 0 1-.095.16l-.85 1.093a.637.637 0 0 1-.626.244.638.638 0 0 1-.335-.16L.237 6.476z' }
          fill='#FFFFFF'
          fillRule='evenodd'
        />
      </svg>
    </StyledCheckboxSvgWrapper>
  );
};

export default CheckboxSvg;
