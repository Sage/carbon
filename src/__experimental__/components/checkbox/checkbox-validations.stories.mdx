import { useState } from "react";
import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";

import { Checkbox, CheckboxGroup } from ".";

<Meta
  title="Experimental/Checkbox/Validations"
  parameters={{ info: { disable: true } }}
/>

## Validations

Validation status can be set by passing `error`, `warning` or `info` prop to the component

Passing a string to these props will display a properly colored border along with a validation icon and tooltip - string value will be displayed as the tooltip message.

Passing a boolean to these props will display only a properly colored border.

For more information check our [Validations](?path=/docs/documentation-validations--page "Validations") documentation page

### As a string applied to single Checkbox

<Preview>
  <Story name="single checkbox - string validation">
    {() =>
      ["error", "warning", "info"].map((type) => (
        <Checkbox
          id={`checkbox_${type}`}
          key={`checkbox-${type}`}
          {...{ [type]: "Message" }}
          label={`Example checkbox (${type})`}
          name={`checkbox-${type}`}
        />
      ))
    }
  </Story>
</Preview>

### As a boolean applied to single Checkbox

<Preview>
  <Story name="single checkbox - boolean validation">
    {() =>
      ["error", "warning", "info"].map((type) => (
        <Checkbox
          id={`checkbox_${type}-boolean`}
          key={`checkbox-${type}-boolean`}
          {...{ [type]: true }}
          label={`Example checkbox (${type})`}
          name={`checkbox-${type}-boolean`}
        />
      ))
    }
  </Story>
</Preview>

### As a string applied to CheckboxGroup

<Preview>
  <Story name="group checkbox - string validation">
    {() =>
      ["error", "warning", "info"].map((type) => (
        <div key={`checkbox-group-${type}`} style={{ marginBottom: 16 }}>
          <CheckboxGroup
            id={`checkbox-group-${type}`}
            name={`checkbox-group-${type}`}
            groupName={`checkbox-group-${type}`}
            {...{ [type]: "Message" }}
            legend={`Group ${type}`}
          >
            {["One", "Two", "Three"].map((label) => (
              <Checkbox
                id={`checkbox_${type}-${label}`}
                key={`checkbox_${type}-${label}`}
                name={`checkbox_${type}-${label}`}
                label={label}
              />
            ))}
          </CheckboxGroup>
        </div>
      ))
    }
  </Story>
</Preview>

### As a boolean applied to CheckboxGroup

<Preview>
  <Story name="group checkbox - boolean validation">
    {() =>
      ["error", "warning", "info"].map((type) => (
        <div
          key={`checkbox-group-${type}-boolean`}
          style={{ marginBottom: 16 }}
        >
          <CheckboxGroup
            id={`checkbox-group-${type}-boolean`}
            name={`checkbox-group-${type}-boolean`}
            key={`checkbox-group-${type}-boolean`}
            groupName={`checkbox-group-${type}-boolean`}
            {...{ [type]: true }}
            legend={`Group ${type}`}
          >
            {["One", "Two", "Three"].map((label) => (
              <Checkbox
                id={`checkbox_${type}-${label}-boolean`}
                key={`checkbox_${type}-${label}-boolean`}
                name={`checkbox_${type}-${label}-boolean`}
                label={label}
              />
            ))}
          </CheckboxGroup>
        </div>
      ))
    }
  </Story>
</Preview>

### Required

You can use the `required` prop to indicate if the field is mandatory.

<Preview>
  <Story name="required">
    {() => {
      const [checked, setChecked] = useState(false);
      return (
        <Checkbox
          label="I agree to the Terms and Conditions"
          name="required"
          required
          checked={checked}
          onChange={(e) => setChecked((checked) => !checked)}
        />
      );
    }}
  </Story>
</Preview>
