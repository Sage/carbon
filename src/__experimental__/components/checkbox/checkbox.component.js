import React from 'react';
import PropTypes from 'prop-types';
import tagComponent from '../../../utils/helpers/tags';
import CheckboxStyle from './checkbox.style';
import CheckableInput from '../checkable-input/checkable-input.component';
import CheckboxSvg from './checkbox-svg.component';
import withValidations from '../../../components/validations/with-validation.hoc';

const Checkbox = ({
  id, label, onChange, onBlur, value, ...props
}) => {
  const inputProps = {
    ...props,
    onChange,
    onBlur,
    labelInline: true,
    inputId: id,
    inputLabel: label,
    inputValue: value,
    inputType: 'checkbox',
    reverse: !props.reverse
  };

  return (
    <CheckboxStyle
      { ...tagComponent('checkbox', props) }
      { ...props }
    >
      <CheckableInput { ...inputProps }>
        <CheckboxSvg />
      </CheckableInput>
    </CheckboxStyle>
  );
};

Checkbox.propTypes = {
  /** Set the value of the checkbox */
  checked: PropTypes.bool,
  /** Toggles disabling of input */
  disabled: PropTypes.bool,
  /** Displays fieldHelp inline with the checkbox */
  fieldHelpInline: PropTypes.bool,
  /** Unique Identifier for the input. Will use a randomly generated GUID if none is provided */
  id: PropTypes.string,
  /** Sets percentage-based input width */
  inputWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** The content of the label for the input */
  label: PropTypes.string,
  /** Sets label alignment - accepted values: 'left' (default), 'right' */
  labelAlign: PropTypes.string,
  /** Sets percentage-based label width */
  labelWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Accepts a callback function which can be used to update parent state on change */
  onChange: PropTypes.func,
  /** Accepts a callback function which is triggered on blur event */
  onBlur: PropTypes.func,
  /** Reverses label and checkbox display */
  reverse: PropTypes.bool,
  /**
   * Set the size of the checkbox to 'small' (16x16 - default) or 'large' (24x24).
   * No effect when using Classic theme.
   */
  size: PropTypes.string,
  /** the value of the checkbox, passed on form submit */
  value: PropTypes.string
};

Checkbox.defaultProps = {
  reverse: false
};

export default withValidations(Checkbox);
