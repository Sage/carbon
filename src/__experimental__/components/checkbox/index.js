export { default } from './checkbox.component';
