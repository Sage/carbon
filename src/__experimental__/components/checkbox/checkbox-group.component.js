import React from 'react';
import PropTypes from 'prop-types';
import tagComponent from '../../../utils/helpers/tags';
import { StyledCheckboxGroup } from './checkbox.style';
import FormField from '../form-field';
import { InputGroupBehaviour } from '../../../__internal__/input-behaviour';

const CheckboxGroup = (props) => {
  const {
    children,
    groupName,
    error,
    warning,
    info
  } = props;

  const groupLabelId = `${groupName}-label`;

  return (
    <InputGroupBehaviour>
      <StyledCheckboxGroup
        aria-labelledby={ groupLabelId }
        role='checkbox'
        error={ error }
        warning={ warning }
        info={ info }
        { ...tagComponent('checkboxgroup', props) }
      >
        <FormField { ...props }>
          {React.Children.map(children, child => React.cloneElement(child, {
            inputName: groupName,
            error: !!error,
            warning: !!warning,
            info: !!info,
            ...child.props
          }))}
        </FormField>
      </StyledCheckboxGroup>
    </InputGroupBehaviour>
  );
};

CheckboxGroup.propTypes = {
  /** The RadioButton objects to be rendered in the group */
  children: PropTypes.node.isRequired,
  /** Specifies the name prop to be applied to each button in the group */
  groupName: PropTypes.string.isRequired,
  /** Indicate that error has occurred
  Pass string to display icon, tooltip and red border
  Pass true boolean to only display red border */
  error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /** Indicate that warning has occurred
  Pass string to display icon, tooltip and orange border
  Pass true boolean to only display orange border */
  warning: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /** Indicate additional information
  Pass string to display icon, tooltip and blue border
  Pass true boolean to only display blue border */
  info: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /** Margin bottom, given number will be multiplied by base spacing unit (8) */
  mb: PropTypes.oneOf([0, 1, 2, 3, 4, 5, 7]),
  /** Spacing between label and a field for inline label, given number will be multiplied by base spacing unit (8) */
  labelSpacing: PropTypes.oneOf([1, 2])
};

export default CheckboxGroup;
