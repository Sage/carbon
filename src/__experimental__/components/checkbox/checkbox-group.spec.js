import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import { css } from 'styled-components';
import TestRenderer from 'react-test-renderer';
import CheckboxGroup from './checkbox-group.component';
import Checkbox from './checkbox.component';
import { StyledCheckboxGroup } from './checkbox.style';
import StyledCheckableInputSvgWrapper from '../checkable-input/checkable-input-svg-wrapper.style';
import Label from '../label';
import LabelStyle from '../label/label.style';
import { assertStyleMatch } from '../../../__spec_helper__/test-utils';
import baseTheme from '../../../style/themes/base';
import Help from '../../../components/help';

const checkboxValues = ['required', 'optional'];
const groupName = 'my-checkbox-group';

function render(props, renderer = TestRenderer.create) {
  const children = checkboxValues.map(value => (
    <Checkbox
      key={ `cKey-${value}` }
      name={ `check-${value}` }
      onChange={ jest.fn() }
      value={ value }
    />
  ));

  return renderer(
    <CheckboxGroup
      name='group-radio-buttons'
      groupName={ groupName }
      label='Test CheckboxGroup Label'
      { ...props }
    >
      {children}
    </CheckboxGroup>
  );
}

function getCheckboxes(wrapper) {
  return wrapper.find(Checkbox);
}

function getInput(wrapper) {
  return wrapper.find('input');
}

describe('CheckboxGroup', () => {
  describe('child RadioButton prop / key mapping', () => {
    const wrapper = render({}, mount);
    const checkboxes = getCheckboxes(wrapper);
    const checkboxArray = checkboxes.getElements();

    describe.each(checkboxArray)('buttons[%#]', (checkbox) => {
      const index = checkboxArray.indexOf(checkbox);

      describe('key / value (both derived from value prop)', () => {
        const expectedValue = checkboxValues[index];
        const expectedKey = `.$cKey-${expectedValue}`;

        it(`sets the value to ${expectedValue}`, () => {
          expect(checkbox.props.value).toEqual(expectedValue);
        });

        it(`sets the key to ${expectedKey}`, () => {
          expect(checkbox.key).toEqual(expectedKey);
        });
      });

      describe('name', () => {
        it('is set using the RadioButtonGroup groupName prop', () => {
          const buttonWrapper = checkboxes.at(checkboxArray.indexOf(checkbox));
          const input = getInput(buttonWrapper).instance();
          const expectedValue = checkboxValues[index];

          expect(input.name).toEqual(`check-${expectedValue}`);
        });
      });

      describe('checkbox initial state', () => {
        it('checked === false', () => {
          expect(checkbox.props.checked).toBe(false);
        });
      });
    });

    describe('change state', () => {
      let checkboxOne = getCheckboxes(wrapper).at(0);
      let checkboxTwo = getCheckboxes(wrapper).at(1);

      const input = getInput(checkboxOne);
      const target = input.instance();

      input.simulate('change', { target });
      wrapper.update();

      checkboxOne = getCheckboxes(wrapper).at(0);
      checkboxTwo = getCheckboxes(wrapper).at(1);

      it('sets checked === true when it is changed', () => {
        expect(checkboxOne.prop('checked')).toBe(true);
        expect(checkboxTwo.prop('checked')).toBe(false);
      });

      it('sets checked === false when the other button is selected', () => {
        const otherInput = getInput(checkboxTwo);
        const otherTarget = otherInput.instance();

        otherInput.simulate('change', { target: otherTarget });
        wrapper.update();

        checkboxOne = getCheckboxes(wrapper).at(0);
        checkboxTwo = getCheckboxes(wrapper).at(1);

        expect(checkboxOne.prop('checked')).toBe(false);
        expect(checkboxTwo.prop('checked')).toBe(true);
      });
    });

    describe('group label', () => {
      const expectedLabelId = `${groupName}-label`;

      it('sets an appropriate label id (derived from groupname)', () => {
        const label = wrapper.find(Label).first();

        expect(label.props().id).toEqual(expectedLabelId);
      });

      it('sets the aria-labelledby attribute to the labelId', () => {
        const buttonGroup = wrapper.find(StyledCheckboxGroup).first();

        expect(buttonGroup.props()['aria-labelledby']).toEqual(expectedLabelId);
      });
    });

    describe('group help label', () => {
      const text = 'Choose an option';

      wrapper.setProps({
        labelHelp: text
      });

      const help = wrapper.find(Help);

      expect(help.prop('children')).toEqual(text);
    });
  });

  describe('styles', () => {
    it('applies the correct Label styles', () => {
      const wrapper = render().toJSON();

      assertStyleMatch(
        {
          cursor: 'default',
          marginBottom: '16px',
          padding: '0'
        },
        wrapper,
        { modifier: css`${`> ${LabelStyle}`}` }
      );
    });

    describe('checkbox group', () => {
      const wrapper = render({}, mount);
      const validationTypes = {
        hasError: { color: baseTheme.colors.error },
        hasWarning: { color: baseTheme.colors.warning },
        hasInfo: { color: baseTheme.colors.info }
      };
      const validationTypesArr = Object.keys(validationTypes);

      describe.each(validationTypesArr)('group[%s]', (type) => {
        beforeEach(() => {
          const props = {
            hasError: false,
            hasWarning: false,
            hasInfo: false
          };
          props[type] = true;

          wrapper.setProps(props);
        });

        it(`${type}=true`, () => {
          expect(wrapper.prop(type)).toBe(true);
        });

        it('has correct color', () => {
          assertStyleMatch({
            border: `1px solid ${validationTypes[type].color}`
          }, wrapper, { modifier: `${StyledCheckableInputSvgWrapper} svg` });
        });
      });
    });
  });
});
