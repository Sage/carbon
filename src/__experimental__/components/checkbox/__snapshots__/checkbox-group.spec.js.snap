// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup renders as expected 1`] = `
.c2 + .c1 {
  margin-top: 16px;
}

.c4 {
  display: block;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c15 {
  cursor: pointer;
  opacity: 0;
  margin: 0;
  position: relative;
  z-index: 999;
}

.c7 {
  color: rgba(0,0,0,0.9);
  display: block;
  font-weight: 600;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 8px;
}

.c13 {
  display: inline-block;
  position: relative;
}

.c10 .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 .c5 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-top: 0;
  width: auto;
}

.c10 .c5 .c19,
.c10 .c5 .c20 {
  color: rgba(0,0,0,0.65);
  vertical-align: middle;
}

.c10 .c5 .c19:hover,
.c10 .c5 .c20:hover,
.c10 .c5 .c19:focus,
.c10 .c5 .c20:focus {
  color: rgba(0,0,0,0.9);
}

.c10 .c17 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
}

.c9 .c12 {
  padding-top: 1px;
}

.c9 .c16 {
  height: 16px;
}

.c9 svg {
  background-color: #FFFFFF;
  border: 1px solid #668592;
}

.c9 .c14,
.c9 svg {
  height: 16px;
  position: absolute;
  padding: 1px;
}

.c9 .c12,
.c9 .c14,
.c9 .c16,
.c9 svg {
  box-sizing: border-box;
  min-width: 16px;
  width: 16px;
}

.c9 .c14:not([disabled]):focus + .c16,
.c9 .c14:not([disabled]):hover + .c16 {
  box-shadow: 0 0 0 3px #FFB500;
}

.c9 .c5 {
  width: auto;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
}

.c9 .c17 {
  margin-left: 16px;
  margin-top: 0;
  padding-left: 8px;
}

.c9 .c20 {
  position: relative;
  display: inline-block;
}

.c9 .c14:checked ~ .c16 svg path {
  fill: rgba(0,0,0,0.90);
}

.c0 .c18::before {
  font-size: 16px;
}

.c0 .c8 {
  padding-top: 12px;
}

.c0 > .c1 > .c5 {
  margin-bottom: 4px;
  vertical-align: middle;
}

.c0 > .c1 > .c5 .c20 {
  display: inline-block;
}

<div
  aria-labelledby="my-checkbox-group-label"
  className="c0"
  data-component="checkboxgroup"
  role="checkbox"
>
  <div
    className="c1 c2"
  >
    <div
      className="c3 c4"
    >
      <div
        className="c5 c6"
        width={30}
      >
        <label
          className="c7"
          data-element="label"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          Test CheckboxGroup Label
        </label>
      </div>
      <div
        className="c8 c9"
        data-component="checkbox"
        name="check-required"
      >
        <div
          className="c10"
          name="check-required"
        >
          <div
            className="c1 c2"
          >
            <div
              className="c3 c11"
            >
              <div
                className="c12 c13"
              >
                <input
                  aria-describedby="cId-required-help"
                  aria-labelledby="cId-required-label"
                  className="c14 c15"
                  id="cId-required"
                  name="check-required"
                  onBlur={[Function]}
                  onChange={[MockFunction]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  role="checkbox"
                  type="checkbox"
                  value="required"
                />
                <div
                  className="c16 "
                >
                  <svg
                    focusable="false"
                    height="10"
                    viewBox="0 0 12 10"
                    width="12"
                  >
                    <path
                      d="M.237 6.477A.752.752 0 0 1 .155 5.47l.851-1.092a.63.63 0 0 1 .934-.088l2.697 1.964, 4.674-6a.63.63 0 0 1 .933-.088l1.015.917c.28.254.317.703.081 1.005L6.353 8.492a.725.725, 0 0 1-.095.16l-.85 1.093a.637.637 0 0 1-.626.244.638.638 0 0 1-.335-.16L.237 6.476z"
                      fill="#FFFFFF"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c8 c9"
        data-component="checkbox"
        name="check-optional"
      >
        <div
          className="c10"
          name="check-optional"
        >
          <div
            className="c1 c2"
          >
            <div
              className="c3 c11"
            >
              <div
                className="c12 c13"
              >
                <input
                  aria-describedby="cId-optional-help"
                  aria-labelledby="cId-optional-label"
                  className="c14 c15"
                  id="cId-optional"
                  name="check-optional"
                  onBlur={[Function]}
                  onChange={[MockFunction]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  role="checkbox"
                  type="checkbox"
                  value="optional"
                />
                <div
                  className="c16 "
                >
                  <svg
                    focusable="false"
                    height="10"
                    viewBox="0 0 12 10"
                    width="12"
                  >
                    <path
                      d="M.237 6.477A.752.752 0 0 1 .155 5.47l.851-1.092a.63.63 0 0 1 .934-.088l2.697 1.964, 4.674-6a.63.63 0 0 1 .933-.088l1.015.917c.28.254.317.703.081 1.005L6.353 8.492a.725.725, 0 0 1-.095.16l-.85 1.093a.637.637 0 0 1-.626.244.638.638 0 0 1-.335-.16L.237 6.476z"
                      fill="#FFFFFF"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
